lib/model/model.go

```diff
@@ -1243,18 +1243,65 @@ func (m *Model) updateLocals(folder string, fs []protocol.FileInfo) {
 		return
 	}
 	files.Update(protocol.LocalDeviceID, fs)
-
+	
 	filenames := make([]string, len(fs))
 	for i, file := range fs {
 		filenames[i] = file.Name
 	}
-
+	
 	events.Default.Log(events.LocalIndexUpdated, map[string]interface{}{
 		"folder":    folder,
 		"items":     len(fs),
 		"filenames": filenames,
 		"version":   files.LocalVersion(protocol.LocalDeviceID),
 	})
+	
+	// Lets us know if file/folder change was originated locally or from a network
+	// sync update.  Now write these to a global log file.
+	if remoteUpdate != true {
+		m.localDiskUpdate(m.folderCfgs[folder].Path(), fs)
+	}
+}
+
+func (m *Model) localDiskUpdate(path string, files []protocol.FileInfo) {
+	// For windows paths, strip unwanted chars from the front
+	path = strings.Replace(path, "\\\\?\\", "", 1) 
+	
+	for _, file := range files {
+		objType := "file"
+		action := "modified"
+		
+		for _, version := range file.Version {
+			if m.shortID == version.ID {
+				// If our local vector is verison 1 AND it is the only version vector so far seen for this file then
+				// it is a new file.  Else if it is > 1 it's not new, and if it is 1 but another shortId version vector
+				// exists then it is new for us but created elsewhere so the file is still not new but modified by us.
+				// Only if it is truely new do we change this to 'added', else we leave it as 'modified'.
+				if version.Value == 1 && len(file.Version) == 1 {
+					action = "added"
+				}
+			}
+		}
```

Comment:
Ok.  I was looping through but it seems our ID will always be the first one if we have seen the file.  So it seems there's no need for **if m.shortID == version.ID {**.

It seems to work file however I did notice a different bug when I tested (happens even on code before I made this change).

Apparently if you created a dir or file that was previously deleted (even a while ago) when you recreate my code says "modified" instead of "added".  On closer inspection it appears that syncthing is remembering the old object with the same name (even though it is technically a different object).  This seems like a bug somewhere else in the code then.  I don't really know how to fix it from within my code.